src/fs/metadata_log.cc

```diff
@@ -101,18 +101,50 @@ void metadata_log::write_update(inode_info::file& file, inode_data_vec new_data_
     } else {
         cut_out_data_range(file, new_data_vec.data_range);
     }
-
     file.data.emplace(new_data_vec.data_range.beg, std::move(new_data_vec));
 }
 
 void metadata_log::cut_out_data_range(inode_info::file& file, file_range range) {
-    file.cut_out_data_range(range, [](inode_data_vec data_vec) {
-        (void)data_vec; // TODO: for compaction: update used inode_data_vec
+    using CR = inode_info::file::cut_result;
+    file.cut_out_data_range(range, [&](inode_data_vec data_vec, CR cr) {
+        int new_writes_cnt;
+        if (cr == CR::BOTH) {
+            new_writes_cnt = 2;
+        } else if (cr == CR::LEFT || cr == CR::RIGHT) {
+            new_writes_cnt = 1;
+        } else {
+            new_writes_cnt = 0;
+        }
+        std::visit(overloaded {
+            [&](inode_data_vec::in_mem_data& mem) {
+                size_t log_decrease = ondisk_entry_size<ondisk_small_write_header>(data_vec.data_range.size());
+                _rewrite_log_size -= log_decrease;
+                _rewrite_log_size += new_writes_cnt * ondisk_entry_size<ondisk_small_write_header>(0);
+            },
+            [&](inode_data_vec::on_disk_data& disk_data) {
+                // TODO: Differentiate between large write and large write without mtime
+                size_t log_decrease = data_vec.data_range.size() == _cluster_size ?
+                    ondisk_entry_size<ondisk_large_write>() :
+                    ondisk_entry_size<ondisk_medium_write>();
+                _rewrite_log_size -= log_decrease;
+                // Remaining data is smaller than original so it's less than large write
+                _rewrite_log_size += new_writes_cnt * ondisk_entry_size<ondisk_medium_write>();
+                _clusters[offset_to_cluster_id(disk_data.device_offset, _cluster_size)].remove_data(disk_data.device_offset, data_vec.data_range);
```

Comment:
`_clusters` this is not a good name it should be corresponding to the name of the struct cluster_info I have elaborated earlier.